code/modules/projectiles/gun.dm

```diff
@@ -308,10 +309,16 @@
 
 	return TRUE
 
-/obj/item/gun/proc/Fire(atom/target, mob/living/user, clickparams, pointblank=0, reflex=0)
+/obj/item/gun/proc/Fire(atom/target, mob/living/user, clickparams, pointblank=0, reflex=0, accuracy_decrease=0, is_offhand=0)
 	if(!fire_checks(target,user,clickparams,pointblank,reflex))
 		return FALSE
 
+	if(!is_offhand && user.a_intent == I_HURT) // no recursion
+		var/obj/item/gun/SG = user.get_inactive_hand()
+		if(istype(SG))
+			var/decreased_accuracy = (SG.w_class * 2) + SG.offhand_accuracy
```

Comment:
Can you maybe explain how the decreased_accuracy calculation works (especially with negative `offhand_accuracy` and whatnot)?

I don't see anything wrong with the code but I just don't completely understand this bit and don't want to approve something I don't understand.

I also wonder how this works with a burst shot? Does just every next shot get increasingly more inaccurate?

It'd also be amazing if you could provide more examples of how many shots hit for different `offhand_accuracy` levels.

Reply:
Makes sense, I checked the values and they seem reasonable.